File: Modules/Common/Equipments/SIE.Web.Equipments/EquipmentCards/Commands/SubmitEquipCardCommand.cs
using SIE.Equipments.EquipmentCards;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Text;

namespace SIE.Web.Equipments.EquipmentCards.Commands
{
    /// <summary>
    /// 设备立卡批量提交
    /// </summary>
    public class SubmitEquipCardCommand : ViewCommand
    {
        /// <summary>
        /// 执行命令
        /// </summary>
        /// <param name="args">参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            List<double> selectedIds = new List<double>(args.SelectedIds);
            RT.Service.Resolve<EquipmentCardController>().SubmitEquipmentCard(selectedIds);
            return true;
        }
    }
}

File: Modules/Common/Equipments/SIE.Web.Equipments/EquipmentCards/Commands/SubmitEquipCardCommand.js
SIE.defineCommand('SIE.Web.Equipments.EquipmentCards.Commands.SubmitEquipCardCommand', {
    meta: { text: "提交", group: "edit", iconCls: "icon-Refuse icon-blue" },
    canExecute: function (view) {
        var selectModels = view.getSelection();
        if (selectModels.length == 0) {
            return false;
        }
        var res = true;
        SIE.each(selectModels, function (model) {
            if (model.data.ApprovalStatus !== SIE.Equipments.Enums.ApprovalStatus.Draft.value
                && model.data.ApprovalStatus !== SIE.Equipments.Enums.ApprovalStatus.Reject.value) {
                res = false;
                return false;
            }
        });
        return res;
    },
    execute: function (view, source) {
        var selectModels = view.getSelection();
        var selectIds = view.getSelectionIds(selectModels);
        SIE.Msg.askQuestion("是否提交？提交后单据不能修改。".t(),
            function () {
                view.execute({
                    withIds: true,
                    selectIds: selectIds,
                    success: function (res) {
                        SIE.Msg.showMessage("提交成功!".t());
                        view.reloadData();
                    }
                });
            });
    }
});